homeassistant/components/zeroconf/__init__.py

```diff
@@ -243,7 +243,7 @@ async def _async_register_hass_zc_service(
 
     _LOGGER.info("Starting Zeroconf broadcast")
     try:
-        await aio_zc.async_register_service(info)
+        await aio_zc.async_register_service(info, allow_name_change=True)
     except NonUniqueNameException:
```

Comment:
Will it still throw an error if it can change the name?

Reply:
Nope.  We can remove the try